### DISCUSSION TITLE
Redis error in auth0 after switching from nginx to Envoy API Gateway

### DISCUSSION THREAD

**Discussion by sfontana9** [mia-platform/community#56] [Q&A] [answered]
Hi, we're switching our project's API Gateway from Nginx to Envoy, and we're facing an issue with Auth0 client.
It looks like Auth0 is not able to verify the authentication of any received token, logging this error 

**[1666026738208] ERROR: failed token retrieval: access token in redis is empty
    reqId: "463bff10-cb41-4e8e-8b9d-f664040e430d"
    error: "access token in redis is empty"**

so all our requests get a 401 UNAUTHORIZED response message.
As additional information, using the same token (generated from /oauth/token through envoy) on the old Nginx API Gateway is working fine.
Any hint on how to fix this? Thank you

**Comment by fredmaggiowski** [comment, marked as answer]
Hi, my guess is that the original host header (`X-Forwarded-Host`) is not correctly forwarded by the API Gateway, some advanced configurations may be required.

Please note that, for other features, Auth0Client also uses the `X-Forwarded-Proto` header so consider forwarding this too it that's the case.
